### PR TITLE
Add `grains` exercise

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

**Comment by axtens** [comment]
1. You need to be specific about which `between` you want, that is which namespace. 
```
ok> help between

d:between (since 22.07)
SED: d1 d2 d3 -- T

Like n:between, but for dates. true if d2 ≤ d1 ≤ d3.

n:between
SED: n low hi -- T

Return true if the number n is between low and hi, inclusive; e.g. if low ≤ n
 ≤ hi.
```
2. In `!if null? ;then` are you testing for null or returning null? If the latter, remove the trailing question mark.
3. `between` will consume your in-coming `n`. You'll need to `dup` it

File: exercises/practice/grains/test.8th [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+"grains on square 64"
+    ( 64 square )
+    9223372036854775808
+    equal?
+
+"square 0 is invalid"
+    ( 0 square )
+    null
+    equal?
```
Comment (glennj) [inline exercises/practice/grains/test.8th:46]:
```suggestion
    null?
```
And the next two tests.

File: exercises/practice/grains/.meta/example.8th [hunk 2]
```diff
@@ -0,0 +1,9 @@
+: square \ n -- n
+  1 64 between !if null? ;then
```
Comment (glennj) [inline exercises/practice/grains/.meta/example.8th:2]: As Bruce mentioned, need a `dup` first. And then, if not between, need to `drop` before adding `null` to the stack.
```suggestion
  dup 1 64 between !if drop null ;then
```
As it stands, the `n:1- 2 swap n:^` is operating on the _expected value_: you _should_ be seeing the "PANIC: expected stack depth to be 3" message. Is your main branch up-to-date (or your local copy of the 8th test runner)?

**Review by glennj — CHANGES_REQUESTED** [review]

File: exercises/practice/grains/.meta/example.8th (hunk 2 above)
Comment (glennj) [reply exercises/practice/grains/.meta/example.8th:2]: Alternately, copy the input to the r-stack:
```forth
: square
  dup>r
  1 64 between !if null ;then
  r> n:1- 2 swap n:^
;
```

**Comment by ErikSchierboom** [comment]
Not sure what happened there. I thought I had ran `bin/test-no-docker -r grains` to verify, but must have fumbled.

**Review by axtens — APPROVED** [review]

**Comment by axtens** [comment]
So have you merged this already, @ErikSchierboom ?

**Comment by ErikSchierboom** [comment]
I haven't because Glenn has requested changes in their review, so I'd like them to confirm I've done the right thing :)

**Review by glennj — APPROVED** [review]
Looks good

**Pull request merged**